index.js

```diff
@@ -60,8 +60,12 @@ function prettyTmpDirName (obj, prop) {
       .replace(/^_+/, '')
   }
 
-  var cleanObjectName = cleanString(obj.constructor && obj.constructor.name)
-  if (cleanObjectName === 'object') cleanObjectName = ''
+  var pluginName = obj.pluginName;
+  var cleanObjectName =
+      cleanString(typeof pluginName === 'string'
+          ? pluginName
+          : (obj.constructor && obj.constructor.name));
```

Comment:
I'm not sure where the dislike for semicolons comes from, but if it's necessary, will take them out